### PR TITLE
Add access token renewal

### PR DESCRIPTION
**Pull request by R0ixy**
Asana - https://app.asana.com/0/1202525679663009/1202879226111242/f

### CODE REVIEW

File: frontend/src/store/queries/index.ts [hunk 1]
```diff
@@ -25,8 +28,49 @@ const baseQuery = fetchBaseQuery({
...
+  await mutex.waitForUnlock();
+  let result = await baseQuery(args, api, extraOptions);
+
+  if (result.error && result.error.status === 401) {
+    if (!mutex.isLocked()) {
+      const release = await mutex.acquire();
+      try {
+        const refreshResult = await baseQuery(
+          {
+            url: '/auth/refreshToken',
```
Comment (oleksandr-danylchenko) [inline frontend/src/store/queries/index.ts]: Please use API object

**Review by oleksandr-danylchenko — CHANGES_REQUESTED** [review]

File: frontend/src/store/queries/index.ts [hunk 2]
```diff
@@ -25,8 +28,49 @@ const baseQuery = fetchBaseQuery({
...
+  await mutex.waitForUnlock();
+  let result = await baseQuery(args, api, extraOptions);
+
+  if (result.error && result.error.status === 401) {
+    if (!mutex.isLocked()) {
+      const release = await mutex.acquire();
+      try {
+        const refreshResult = await baseQuery(
+          {
+            url: `${ApiPath.AUTH}/refreshToken`,
```
Comment (oleksandr-danylchenko) [inline frontend/src/store/queries/index.ts]: `/refreshToken` also should be part of the `ApiPath`

**Review by oleksandr-danylchenko — CHANGES_REQUESTED** [review]

File: frontend/src/store/queries/index.ts (hunk 2 above)
Comment (R0ixy) [reply frontend/src/store/queries/index.ts]: ouch, ok

File: frontend/src/common/enums/api/api-path.enum.ts [hunk 3]
```diff
@@ -1,5 +1,6 @@
 enum ApiPath {
   AUTH = '/auth',
+  REFRESH_TOKEN = '/refreshToken',
```
Comment (oleksandr-danylchenko) [inline frontend/src/common/enums/api/api-path.enum.ts]: These paths should include the full URL, so the `REFRESH_TOKEN` should be:
```suggestion
  AUTH = '/auth',
  REFRESH_TOKEN = '/auth/refreshToken',
```

**Review by oleksandr-danylchenko — CHANGES_REQUESTED** [review]

**Review by oleksandr-danylchenko — APPROVED** [review]

**Pull request merged**